src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
```php
$this->viewConfig()->layout('admin');  // Can't name view() as it would clash with action name
``` 
looks sounds good to me.

The class could be `Cake\View\ViewConfig` and method to get proxy instance could be `Cake\View\ViewVarsTrait::viewConfig()`.